### PR TITLE
Add proposal for sending urls to the web page of an item.

### PR DESCRIPTION
**Pull request by WouterSioen**

### CODE REVIEW

File: apiary.apib
Comment (mrtus) [inline apiary.apib]: Q: Should we start using meta data as a root for these? For me this is meta data and does not belong to the model itself.

Comment (WouterSioen) [reply apiary.apib]: added_at and updated_at is also metadata imo

Comment (mrtus) [reply apiary.apib]: The model can know about its own details, like creation & update times. (Domain)
It should/can not know its own location in the application. (Application)

TBD offcourse :) that is why this is a proposal right? 😉

File: src/02-crm/contacts.apib
Comment (stivni) [inline src/02-crm/contacts.apib]: PP : not sure if "canonical" makes sense in this context

Isn't his just the "application_url", "visit_url", "web_url", "page_url" or something?

File: apiary.apib
Comment (WouterSioen) [reply apiary.apib]: Yes, we kind of need this implemented, but aren't sure yet where to put it. Summoning @stivni and @jenssegers here.

File: src/02-crm/contacts.apib
Comment (WouterSioen) [reply src/02-crm/contacts.apib]: Canonical is wat Dietrich suggested based on what Drupal names this, but canonical probably references to the SEO meta tag "canonical_url" in Drupal.

Any preference?

Comment (stivni) [reply src/02-crm/contacts.apib]: B: I thought we agreed on adding the url in a meta-data-like property either in 

1. `data: { }, metaData : { url: ... }`
2. `data: { _url: ... }`
3. `data: { @url: ... }`

And do we wrap inside a `links` attribute?

```
links {
    "self": "https://app.teamleader.eu/foo/bar/baz"
}
```

Comment (stivni) [reply src/02-crm/contacts.apib]: @jenssegers thoughts?

Comment (WouterSioen) [reply src/02-crm/contacts.apib]: I know we're not doing REST, but in REST there's the concept of HATEOS where all links are under a `links` property which all have a rel (relationship) and a href (the actual link). This is f.e. also used for pagination.

Comment (mrtus) [reply src/02-crm/contacts.apib]: @stivni As mentioned here: https://github.com/teamleadercrm/api/pull/195#discussion_r204417716 I do like option 1 in combination with the `links` with maybe a more explicit url name like you mentioned in your first comment 👍

Comment (jenssegers) [reply src/02-crm/contacts.apib]: Not sure if it belongs in the (REST) `links` section, that's more used for navigating through relations or pagination.

I don't know if `canonical_url` is the correct name, but I kinda like it. We just need to make sure this is actually the correct term our url.

I would choose to either go for the currently proposed:

```
data: {
  id: 123,
  canonical_url: "https://app.teamleader.eu/foo/bar/baz"
}
```

Or by marking it somehow as meta data:

```
data: {
  id: 123,
  @canonical_url: "https://app.teamleader.eu/foo/bar/baz"
}
```

With my main preference going to the first one.

I'm just afraid that once we start adding meta data to objects by prepending it with a `_` or `@`, we will end up with endless discussions about what belongs in meta data and what not.

File: apiary.apib
Comment (WouterSioen) [reply apiary.apib]: @mrtus we can't really differentiate between a `meta` and a `data` object anymore, since in `.list` endpoints, all data is returned as an array in the `data` key.

We could put an array `meta` in the `data` itself, but this doesn't give a lot of advantages imo.

 :ok: by @WouterSioen on 25/07/2018 15:13:03

File: src/02-crm/contacts.apib
Comment (WouterSioen) [reply src/02-crm/contacts.apib]: We went through a discussion and checked how major apis handled this. We came to the conclusion that `web_url` in the `data` object would be the most logical solution. (a bit more information in the commit message).

**Review by dietrichm — APPROVED** [review]

**Comment by WouterSioen** [comment]
@stivni @jenssegers can we get this merged if we don't get a better proposal?

**Comment by thomasnoe07** [comment]
@WouterSioen @jenssegers @stivni so to be merged or not that is the question here :-) If we do not merge it lets close this PR?

**Comment by WouterSioen** [comment]
Let’s merge it instead. It’s implemented behind a feature flag which can easily be rolled out. Just want a final approval here before shipping

**Comment by dietrichm** [comment]
Superseded by https://github.com/teamleadercrm/api/pull/225.

**Pull request closed without merging**
